packages/material-ui-system/src/css.js

```diff
@@ -13,15 +13,32 @@ function omit(input, fields) {
   return output;
 }
 
+function handleCss(styleFunction, theme) {
+  const apply = styles => {
+    const cssStyles = omit(styles, styleFunction.filterProps);
+    const output = {
+      ...styleFunction({ theme, ...styles }),
+      ...cssStyles,
+    };
+
+    Object.keys(cssStyles).forEach(key => {
+      if (typeof cssStyles[key] === 'object') {
+        output[key] = apply(cssStyles[key]);
+      }
+    });
+
+    return output;
+  };
+
+  return apply;
+}
+
 function css(styleFunction) {
   const newStyleFunction = props => {
-    const output = styleFunction(props);
+    const output = styleFunction(props) || {};
 
     if (props.css) {
-      return {
-        ...merge(output, styleFunction({ theme: props.theme, ...props.css })),
-        ...omit(props.css, [styleFunction.filterProps]),
```

Comment:
I believe this is wrong? `styleFunction.filterProps` is already an array right?
But this actually seemed to save the tests, because `output` seems to be `null`, but since the omit fails it includes the whole object.

I added a null check to `output` to solve this, not sure if thats the best solution.

This is what currently happens (?):
```
{
  ...null, // This should have { color } in the tests
  ...props.css // This should include everything but { color } in the tests
}
```

Reply:
Could you add a test in a separate PR that better explains what this line was/is for?